MPMap.html

```diff
@@ -0,0 +1,241 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="UTF-8" />
+    <title>My Climbs</title>
+    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.9.4/dist/leaflet.css"
+        integrity="sha256-p4NxAoJBhIIN+hmNHrzRCf9tD/miZyoHS5obTRR9BMY="
+        crossorigin=""/>
+     <script src="https://unpkg.com/leaflet@1.9.4/dist/leaflet.js"
+        integrity="sha256-20nQCchB9co0qIjJZRGuk2/Z9VM+kNiyxNV1lvTlZBo="
+        crossorigin=""></script>
+        <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script> 
+
+
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <style>#map { height: 800px; width: 800px }</style>
+</head>
+<body>
+    <h3>Enter your mountain Project url</h5>
+    <input type="text" id="mpUser">
+    <button onclick="getProfile()">Submit</button>
+    <br><br>
+    <div id="map"></div>
+
+    <script>
+        let routeUrls = [];
+        let areaUrls = [];
+        let currentLat;
+        let currentLong;
+        let routeNameArr = [];
+        let gpsCoords = [];
+        let mark;
+        let markArray = [];
+
+        // Climb grades (5.9, WI4, etc.)
+        let rockGrade;
+        let iceGrade;
+        let mixedGrade;
+        let aidGrade;
+        let aidRating = [];
+
+        for(let k = 0; k < 6; k++){
+            aidRating.push("A" + k, "C" + k);
+        }
+
+        // Layers for markers and tooltips of climbs 
+        let rockLayer = L.layerGroup();
+        let iceLayer = L.layerGroup();
+        let mixedLayer = L.layerGroup();
+        let aidLayer = L.layerGroup();
+
+        // Arrays to hold grades of climbs
+        let rockDifficulty = [];
```

Comment:
What if we defined a class like:
```
class Route {
    type; // Mixed, Rock, Ice
    name;
    grade; // 5.3, WI4
    danger; // PG13, R, X
}
```
Then when you parse the profile create an array of `Routes` and simply return that. Now you can loop through your array and do something like:

```
switch (route.type) {
    case 'Ice': // add to ice layer
    case 'Mixed': // add to mixed layer
    default: // add to rock layer
}
```